### PR TITLE
feat: New works from galleries you follow home page rail

### PR DESCRIPTION
**Pull request by olerichter00**
Addresses [ONYX-48]

## Description

This PR adds the new "New Works From Galleries You Follow" rail to the home screen (for logged-in users only) and creates a new page under `/new-works-from-galleries-you-follow` ([matches route in the app](https://github.com/artsy/eigen/blob/858f27a5bcddc627fa1673c9f1de5d8a25de5edf/src/app/routes.ts#L227)).

## Screenshots

**Rail**

Screenshot 2023-09-05 at 15 43 02

**Page**

Screenshot 2023-09-06 at 17 02 57

[ONYX-48]: https://artsyproduct.atlassian.net/browse/ONYX-48?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Home/Components/HomeNewWorksFromGalleriesYouFollowRail.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        trackEvent(trackingEvent)
+      }}
+      getItems={() => {
+        return artworks.map(artwork => (
+          <ShelfArtworkFragmentContainer
+            artwork={artwork}
+            key={artwork.internalID}
+            lazyLoad
+            // TODO: add troveArtworksRail to the union type of auth context module
+            // @ts-ignore
```
Comment (olerichter00) [inline src/Apps/Home/Components/HomeNewWorksFromGalleriesYouFollowRail.tsx:61]: https://github.com/artsy/cohesion/pull/451

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Routes added to `routes.tsx` should have a corresponding cypress.js smoke test. See the `cypress/e2e` folder for examples.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 82aaa569b2f726e040de6c845cac1b53f63f304d
</p>

File: src/Apps/Home/Components/HomeNewWorksFromGalleriesYouFollowRail.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      viewAllLabel="View All Works"
+      viewAllHref="/new-works-from-galleries-you-follow"
+      viewAllOnClick={() => {
+        const trackingEvent: ClickedArtworkGroup = {
+          action: ActionType.clickedArtworkGroup,
+          context_module: ContextModule.newWorksByGalleriesYouFollowRail,
+          context_page_owner_type: OwnerType.home,
+          destination_page_owner_type: OwnerType.collection,
+          destination_page_owner_id: "932d0b13-3cf1-46d1-8e49-18b186230347",
+          destination_page_owner_slug: "curators-picks-emerging",
```
Comment (dzucconi) [inline src/Apps/Home/Components/HomeNewWorksFromGalleriesYouFollowRail.tsx:49]: This doesn't match the slug it actually points to?

File: src/Apps/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollowApp.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        pathname="/new-works-from-galleries-you-follow"
+      />
+
+      <Spacer y={4} />
+
+      <Text variant="xl">New Works from Galleries You Follow</Text>
+
+      <Spacer y={4} />
+
+      <LogInPrompt
```
Comment (dzucconi) [inline src/Apps/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollowApp.tsx:52]: This component shouldn't have an external margin.

File: src/Apps/NewWorksFromGalleriesYouFollow/__tests__/NewWorksFromGalleriesYouFollowApp.jest.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      Me: () => ({
+        newWorksFromGalleriesYouFollowConnection: null,
+      }),
+    })
+
+    expect(screen.getByText("Nothing yet.")).toBeInTheDocument()
+  })
+})
+
+const meResponse = {
```
Comment (dzucconi) [inline src/Apps/NewWorksFromGalleriesYouFollow/__tests__/NewWorksFromGalleriesYouFollowApp.jest.tsx:82]: This seems excessive. Just autogenerate everything except for the relevant bits.

File: src/Apps/Home/__tests__/HomeNewWorksFromGalleriesYouFollowRail.jest.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        destination_page_owner_id: "<Artwork-mock-id-1>",
+        destination_page_owner_slug: "<Artwork-mock-id-2>",
+        destination_page_owner_type: "artwork",
+        type: "thumbnail",
+      })
+    })
+  })
+})
+
+const meResponse = {
```
Comment (damassi) [inline src/Apps/Home/__tests__/HomeNewWorksFromGalleriesYouFollowRail.jest.tsx:89]: Ditto here, re generating responses. Why create fixtures when they can be autogenerated by relay's testing tools? We should only create fixtures out of the fields that are needed directly within the test, or for things that help the component render in certain ways.

File: src/routes.tsx [hunk 6]
```diff
@@ -99,6 +100,7 @@ export const getAppRoutes = (): AppRouteConfig[] => {
     { routes: myCollectionInsightsRoutes },
     { routes: myCollectionInsightsCollectorProfileRoutes },
     { routes: newForYouRoutes },
+    { routes: newWorksFromGalleriesYouFollowRoutes },
```
Comment (damassi) [inline src/routes.tsx:103]: There's a danger warning around adding a smoke test that needs to be resolved before this is merged (unless its strictly logged-in-only and we can't)

File: src/Apps/Home/Components/HomeNewWorksFromGalleriesYouFollowRail.tsx (hunk 2 above)
Comment (olerichter00) [reply src/Apps/Home/Components/HomeNewWorksFromGalleriesYouFollowRail.tsx]: Thanks, I've fixed the tracking.

File: src/Apps/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollowApp.tsx (hunk 3 above)
Comment (olerichter00) [reply src/Apps/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollowApp.tsx:54]: Not sure what you mean by external margin 🤔 

I've copied the code from https://www.artsy.net/new-for-you which shows the component the same way.

File: src/Apps/NewWorksFromGalleriesYouFollow/__tests__/NewWorksFromGalleriesYouFollowApp.jest.tsx (hunk 4 above)
Comment (olerichter00) [reply src/Apps/NewWorksFromGalleriesYouFollow/__tests__/NewWorksFromGalleriesYouFollowApp.jest.tsx]: Removed the fixture 👍

File: src/Apps/Home/__tests__/HomeNewWorksFromGalleriesYouFollowRail.jest.tsx (hunk 5 above)
Comment (olerichter00) [reply src/Apps/Home/__tests__/HomeNewWorksFromGalleriesYouFollowRail.jest.tsx]: Removed the fixture 👍

File: src/routes.tsx (hunk 6 above)
Comment (olerichter00) [reply src/routes.tsx:103]: It's strictly logged-in only, but I've added a smoke test anyway to ensure that the page renders (only checking the title) when opening the URL.

File: src/Apps/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollowApp.tsx (hunk 3 above)
Comment (dzucconi) [reply src/Apps/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollowApp.tsx:54]: It's not part of this PR but this `LogInPrompt` component has a size 4 margin hanging off the bottom of it. This makes it both confusing to use and brittle. https://github.com/artsy/force/blob/main/docs/best_practices.md#avoid-external-margins

**Review by dzucconi — DISMISSED** [review]
LGTM

**Pull request merged**